lms/djangoapps/grades/rest_api/v1/tests/test_gradebook_views.py

```diff
@@ -738,6 +738,12 @@ def test_gradebook_data_filter_username_contains(self, login_method):
                 self.assertIsNone(actual_data['previous'])
                 self.assertEqual(expected_results, actual_data['results'])
 
+                # the number/type of users in a course depends on the login_method
```

Comment:
It's not super obvious to me why this would be the case. If you dug in, could you expand the comment? Seems spooky

Reply:
The [`login_course_admin`](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/rest_api/v1/tests/test_gradebook_views.py#L339) and [`login_course_staff`](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/rest_api/v1/tests/test_gradebook_views.py#L329) methods create a CourseEnrollment for the user and course, but the [`login_staff`](https://github.com/edx/edx-platform/blob/master/lms/djangoapps/grades/rest_api/v1/tests/test_gradebook_views.py#L321) method doesn't, so when the test runs with `login_method` equal to either `login_course_admin` or `login_course_staff`, there's an extra enrollment as compared to `login_staff`.

I asked Simon whether adding a CourseEnrollment to the `login_staff` case would affect the access control functionality of the gradebook API, and he said that it wouldn't, but he wants the use of global staff to be in line with the expectation that a CourseEnrollment is not required for a global staff user to have access to the gradebook API.

I hope that helps to explain it. I switched to use `ddt` parameters instead, which cleans the code up a bit but also makes it harder for me to leave an inline comment. Does this look okay to you?